### PR TITLE
Handle responses not containing JSON (HTML responses)

### PR DESCRIPTION
**Pull request by cedel1**

### CODE REVIEW

File: pycsob/utils.py [hunk 1]
```diff
@@ -16,6 +17,10 @@
     from datetime import datetime
 
 
+class CsobCommunicationError(Exception):
```
Comment (matllubos) [inline pycsob/utils.py]: Can exists some parent of all CSOB exceptions?

File: pycsob/utils.py [hunk 2]
```diff
@@ -72,7 +77,11 @@ def dttm(format_='%Y%m%d%H%M%S'):
 def validate_response(response, key):
     response.raise_for_status()
 
-    data = response.json()
+    try:
+        data = response.json()
+    except JSONDecodeError:
```
Comment (matllubos) [inline pycsob/utils.py:75]: what other response errors?

File: pycsob/utils.py (hunk 1 above)
Comment (cedel1) [reply pycsob/utils.py]: Currently, there's no other CSOB exception in the code.  Do you want me to make one general CSOB exception?

File: pycsob/utils.py (hunk 2 above)
Comment (cedel1) [reply pycsob/utils.py:75]: Added handling of other response errors, HTTP errors here and connection related errors in the HTTPAdapter.

Comment (cedel1) [reply pycsob/utils.py:75]: Better?

File: pycsob/utils.py (hunk 1 above)
Comment (cedel1) [reply pycsob/utils.py]: Moved to a separate file and inherited from baseCSOB Exception.

**Review by Formulka — APPROVED** [review]

File: pycsob/utils.py (hunk 2 above)
Comment (matllubos) [reply pycsob/utils.py:75]: no use RequestException not HTTPError. There are more requests exceptions for example Timeout, Proxy, SSL

Comment (cedel1) [reply pycsob/utils.py:75]: I don't understand.

The documentation clearly states:
"Response.raise_for_status() will raise an HTTPError if the HTTP request returned an unsuccessful status code."
and
"All exceptions that Requests explicitly raises inherit from requests.exceptions.RequestException."

which is supported by the source code at https://docs.python-requests.org/en/latest/_modules/requests/exceptions/#RequestException and requests own code at requests/adapters.py

I check for HTTPError after `raise_for_status` call and for `RequestException` in the HTTPAdapter when any outside call through the adapter is made (as mentioned above). As I understand it, that should cover all requests exceptions, except those, that `requests` themselves will not catch - but the ones you mention are caught and covered.

What exactly do you want me to do?

**Pull request merged**